### PR TITLE
[EN-324] Implements the multiStore on commercetools

### PR DESCRIPTION
**Pull request by bloodf**
### Related Issues

closes #[EN-261](https://vsf.atlassian.net/browse/EN-261)
closes #[EN-357](https://vsf.atlassian.net/browse/EN-357)
closes #[EN-356](https://vsf.atlassian.net/browse/EN-356)

### Short Description of the PR
this PR applies the multi-store on the CommerceTools, and adds a new Component on the `nuxt-theme-module` package to handle the language and store changes. 

In this PR there are changes on the api-client to match the CommerceTools APIs on the multi-store, when there are requirements for the store ID.

### Screenshots of Visual Changes before/after (if There Are Any)

image

### Pull Request Checklist

- [ ] I have updated the Changelog ([V1](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md)) [v2](https://docs-next.vuestorefront.io/contributing/creating-changelog.html) and mentioned all breaking changes in the public API.
- [ ] I have documented all new public APIs and made changes to existing docs mentioning the parts I've changed so they're up to date.
- [ ] I have tested my Pull Request on production build and (to my knowledge) it works without any issues

- I tested manually my code and it works well with both:
- [x] Default Theme
- [ ] Capybara Theme
- [ ] I have written test cases for my code

- [x] I have followed [naming conventions](https://github.com/kettanaito/naming-cheatsheet)

### CODE REVIEW

File: packages/commercetools/composables/src/useStore/factoryParams.ts [hunk 1]
```diff
@@ -10,7 +10,7 @@ async function load (context: Context, params: UseStoreFactoryLoadParamArguments
 
 // Change param
 async function change (context: Context, { store }: UseStoreFactoryChangeParamArguments) {
-  context.$ct.config.storeService.changeCurrentStore(store.id);
+  context.$ct.config.storeService.changeCurrentStore(`${store.key}`);
```
Comment (andrzejewsky) [inline packages/commercetools/composables/src/useStore/factoryParams.ts:13]: ok you are setting the current store in the config... but then you are reloading the page... so what will happen? i think the default store will be loaded from the config, not the one you just set

Comment (andrzejewsky) [reply packages/commercetools/composables/src/useStore/factoryParams.ts:13]: i don't remember - is there any cookie storing underneath?

Comment (alefbarbeli) [reply packages/commercetools/composables/src/useStore/factoryParams.ts:13]: The `changeCurrentStore` service handle the cookie storage.

Here on line 32:
https://github.com/vuestorefront/vue-storefront/blob/EN-356-implement-useStore-commercetools/packages/commercetools/composables/nuxt/plugin.js

File: packages/commercetools/composables/__tests__/useCart/useCart.spec.ts [hunk 2]
```diff
@@ -73,7 +74,7 @@ describe('[commercetools-composables] useCart', () => {
     const currentCart = { id: 1, version: 1 };
     const response = await clear(context, { currentCart });
 
-    expect(response).toEqual(currentCart);
+    expect(response).toEqual('some cart');
```
Comment (Baroshem) [inline packages/commercetools/composables/__tests__/useCart/useCart.spec.ts:77]: What do you think about storing `some cart` in some kind of a variable? I can see that it is an assertion in several places. Lines 77, 68, 55, 46, 42, 36, 14, 10, 9, 8, 7. Then we could do i.e:
```ts
const MOCKED_CART = 'some cart';

applyCartCoupon: jest.fn(() => ({ data: { cart: MOCKED_CART } }))
expect(response).toEqual(MOCKED_CART);
```

File: packages/commercetools/theme/components/StoreLocaleSelector.vue [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+
+    return {
+      load,
+      changeStore,
+      response,
+      availableStores,
+      selectedStore,
+      isStoreSelected,
+      getStoreLocale,
+      //
```
Comment (Baroshem) [inline packages/commercetools/theme/components/StoreLocaleSelector.vue:107]: Do we need this?

Comment (bloodf) [reply packages/commercetools/theme/components/StoreLocaleSelector.vue]: those 3 are being used in the theme section :)

**Comment by igorwojciechowski** [comment]
Please verify if it's a valid bug related to this issue

Store labels are being changed to language corresponding to selected locale only after selecting the store as well. I think there should be no need to click on a store for labels to be translated/changed to currently selected locale language.

https://user-images.githubusercontent.com/26794636/126326370-c5e11955-da87-4451-b405-907547a12f48.mp4

**Comment by igorwojciechowski** [comment]
Icons are not being displayed if channels are linked to the store.

image

File: packages/commercetools/composables/__tests__/useCart/useCart.spec.ts (hunk 2 above)
Comment (alefbarbeli) [reply packages/commercetools/composables/__tests__/useCart/useCart.spec.ts]: @Baroshem I've done it! Re-review please??

**Comment by igorwojciechowski** [comment]
Minor bug:
When there are no stores created on the backend, the "Choose Language" header is displayed twice.

image

**Comment by alefbarbeli** [comment]
> Minor bug:
> When there are no stores created on the backend, the "Choose Language" header is displayed twice.
> 
> image

fixed duplicated header
image

**Comment by igorwojciechowski** [comment]
Retested the issue after bugfixing - seems ok now!

File: packages/commercetools/theme/components/AppHeader.vue [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+    const result = ref(null);
+
+    const cartTotalItems = computed(() => {
+      const count = cartGetters.getTotalItems(cart.value);
+      return count ? count.toString() : null;
+    });
+
+    const accountIcon = computed(() => isAuthenticated.value ? 'profile_fill' : 'profile');
+
+    loadUser();
```
Comment (filipsobol) [inline packages/commercetools/theme/components/AppHeader.vue:145]: Is there a reason this is not wrapped in `onSSR`?

Comment (alefbarbeli) [reply packages/commercetools/theme/components/AppHeader.vue:145]: It was copy-pasted from `nuxt-theme-module` we just changed the `LocaleSelector` to `StoreLocaleSelector` which is our new component for selecting stores

**Review by Baroshem — APPROVED** [review]
Well done!

**Review by lukaszjedrasik — APPROVED** [review]

**Comment by igorwojciechowski** [comment]
Minor bug:

Store with the same channel set as a both supply and distribution channel is being displayed twice on frontend.

image

**Pull request merged**